db/open_js.go

```diff
@@ -3,13 +3,39 @@
 package db
 
 import (
+	"errors"
+	"syscall/js"
+	"time"
+
+	log "github.com/sirupsen/logrus"
 	"github.com/syndtr/goleveldb/leveldb"
 	"github.com/syndtr/goleveldb/leveldb/storage"
 )
 
-// Open creates a new database using in-memory storage. The given file path is
-// ignored.
+const (
+	// browserFSLoadCheckInterval is frequently to check whether browserFS is
+	// loaded.
+	browserFSLoadCheckInterval = 50 * time.Millisecond
+	// browserFSLoadTimeout is how long to wait for BrowserFS to finish loading
+	// before giving up.
+	browserFSLoadTimeout = 5 * time.Second
```

Comment:
Could the length of time it takes to load `browserFS` be influenced by a users internet speed? If so, any harm in setting this wait time higher?

Reply:
If you are using the `@0x/mesh-browser` package, then no, it will not be affected by your internet speed. The Wasm and JavaScript code will be bundled together and _downloaded_ at the same time. How long it takes to _parse and execute_ the Wasm and JavaScript code could potentially be influenced by your CPU speed. However, it is far more likely that the JavaScript code will execute first due to the fact that the Wasm bytecode is much larger and typically takes longer to parse.

I have considered the idea of offering an alternative way for users to run Mesh in the browser, which is to bundle the Wasm and JavaScript code separately. This would allow use of the [streaming API for WebAssembly](https://www.w3.org/TR/wasm-web-api-1/#streaming-modules) and could potentially improve performance. But for now it's not something I think we should worry about.

The downside to increasing the timeout is that it takes us longer to discover errors related to BrowserFS failing to load.